main/utils.zkasm

```diff
@@ -990,122 +990,306 @@ invalidCall:
-    B                           :MSTORE(numBlocks)
-    ${A & 0x03} => D
-    D                           :MSTORE(leftBytes)
-
-    ; since D is assured to be less than 0x04
-    ; it is enforced that [b7, b6, ..., b1] are 0 since there is no value multiplied by 4
-    ; that equals the field
-    ; Since b0 is assured to be less than 32 bits, b0 * 4 + d0 could not overflow the field
-    B * 4 + D                   :ASSERT
-    D => A
-    0x04 => B
-    $                           :LT,JMPNC(failAssert)
-    0 => B
+                            :JMP(@readPush+E)    ;0
+readPush1:
+    PC => HASHPOS           :JMP(_readPush1)     ;1
+readPush2:
+    PC => HASHPOS           :JMP(_readPush2)     ;2
+readPush3:
+    PC => HASHPOS           :JMP(_readPush3)     ;3
+readPush4:
+    PC => HASHPOS           :JMP(_readPush4)     ;4
+readPush5:
+    PC => HASHPOS           :JMP(_readPushX)     ;5
+readPush6:
+    PC => HASHPOS           :JMP(_readPushX)     ;6
+readPush7:
+    PC => HASHPOS           :JMP(_readPushX)     ;7
+readPush8:
+    PC => HASHPOS           :JMP(_readPushX)     ;8
+readPush9:
+    PC => HASHPOS           :JMP(_readPushX)     ;9
+readPush10:
+    PC => HASHPOS           :JMP(_readPushX)     ;10
+readPush11:
+    PC => HASHPOS           :JMP(_readPushX)     ;11
+readPush12:
+    PC => HASHPOS           :JMP(_readPushX)     ;12
+readPush13:
+    PC => HASHPOS           :JMP(_readPushX)     ;13
+readPush14:
+    PC => HASHPOS           :JMP(_readPushX)     ;14
+readPush15:
+    PC => HASHPOS           :JMP(_readPushX)     ;15
+readPush16:
+    PC => HASHPOS           :JMP(_readPushX)     ;16
+readPush17:
+    PC => HASHPOS           :JMP(_readPushX)     ;17
+readPush18:
+    PC => HASHPOS           :JMP(_readPushX)     ;18
+readPush19:
+    PC => HASHPOS           :JMP(_readPushX)     ;19
+readPush20:
+    PC => HASHPOS           :JMP(_readPushX)     ;20
+readPush21:
+    PC => HASHPOS           :JMP(_readPushX)     ;21
+readPush22:
+    PC => HASHPOS           :JMP(_readPushX)     ;22
+readPush23:
+    PC => HASHPOS           :JMP(_readPushX)     ;23
+readPush24:
+    PC => HASHPOS           :JMP(_readPushX)     ;24
+readPush25:
+    PC => HASHPOS           :JMP(_readPushX)     ;25
+readPush26:
+    PC => HASHPOS           :JMP(_readPushX)     ;26
+readPush27:
+    PC => HASHPOS           :JMP(_readPushX)     ;27
+readPush28:
+    PC => HASHPOS           :JMP(_readPushX)     ;28
+readPush29:
+    PC => HASHPOS           :JMP(_readPushX)     ;29
+readPush30:
+    PC => HASHPOS           :JMP(_readPushX)     ;30
+readPush31:
+    PC => HASHPOS           :JMP(_readPushX)     ;31
+readPush32:
+    PC => HASHPOS           :JMP(_readPush32)    ;32
+
+_readPush1:
+    %MAX_CNT_STEPS - STEP - 2   :JMPN(outOfCountersStep)
+
+    $ => E                      :MLOAD(contractHashId)
+    $ => E                      :HASHP1(E),RETURN
+
+_readPush2:
+    %MAX_CNT_STEPS - STEP - 7   :JMPN(outOfCountersStep)
```

Comment:
Should be 4 steps to follow other functions logic

Reply:
updated to 4 steps.
:smile: